spec_parser/owl_to_context.py

```diff
@@ -0,0 +1,108 @@
+# SPDX-FileCopyrightText: 2023 spdx contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+import json
+from os import path
+
+PROPERTIES_WITH_ENUM_RANGE = [
+    "safetyRiskAssessment",
+    "sensitivePersonalInformation",
+    "annotationType",
+    "externalIdentifierType",
+    "externalReferenceType",
+    "algorithm",
+    "scope",
+    "profile",
+    "completeness",
+    "relationshipType",
+    "confidentialityLevel",
+    "datasetAvailability",
+    "decisionType",
+    "justificationType",
+    "catalogType",
+    "conditionality",
+    "sbomType",
+    "softwareLinkage",
+    "purpose",
+]
+
+REFERENCE_PROPERTY_TYPES = [
+    "core:Element",
+    "core:Agent",
+]
+
+LOCAL_PROFILE_CONTEXT = {
+    "core": "https://spdx.org/rdf/Core/ProfileIdentifierType/core",
+    "software": "https://spdx.org/rdf/Core/ProfileIdentifierType/software",
+    "licensing": "https://spdx.org/rdf/Core/ProfileIdentifierType/licensing",
+    "security": "https://spdx.org/rdf/Core/ProfileIdentifierType/security",
+    "build": "https://spdx.org/rdf/Core/ProfileIdentifierType/build",
+    "ai": "https://spdx.org/rdf/Core/ProfileIdentifierType/ai",
+    "dataset": "https://spdx.org/rdf/Core/ProfileIdentifierType/dataset",
+    "usage": "https://spdx.org/rdf/Core/ProfileIdentifierType/usage",
+    "extension": "https://spdx.org/rdf/Core/ProfileIdentifierType/extension",
+}
+
+
+def convert_spdx_owl_to_jsonld_context(spdx_owl: str, out_dir: str):
+    # spdx_owl must point to the OWL in json-ld format
+    with open(spdx_owl, "r") as infile:
+        owl_dict = json.load(infile)
+
+    context_dict = owl_dict["@context"]
+
+    for node in owl_dict["@graph"]:
+        node_type = node.get("@type")
+        if not node_type:
+            continue
+
+        if "owl:NamedIndividual" in node_type:
+            continue
+
+        node_id = node["@id"]
+        name = get_name_from_node_id(node_id)
+
+        if node_type in ["owl:DatatypeProperty", "owl:ObjectProperty"]:
+            type_id = node["rdfs:range"]["@id"]
+
+            if name in context_dict and context_dict[name]["@id"].startswith("core"):
+                # if in doubt, prioritize core properties
+                continue
+
+            if name in PROPERTIES_WITH_ENUM_RANGE:
+                if name == "profile":
```

Comment:
I think this assumes that the value of `profile` is `core` instead of the IRI of Core profile/namespace.

Reply:
This assumes that the correct IRI for `core` under the `profile` property is of the Enum `ProfileIdentiferType` and therefore must expand to `https://spdx.org/rdf/Core/ProfileIdentifierType/core` and not `https://spdx.org/rdf/v3/Core/`.
Please correct me if that is not true.

Reply:
This will have to change, but let's take the model first and then we revisit this one.